### PR TITLE
Isolate Dropwizard and HK2 dependencies

### LINKED ISSUE #521: Isolate dependencies on Dropwizard + hk2
### Is this a possible security vulnerability?

- [X] This is NOT a possible security vulnerability

### Describe the bug

Dropwizard + hk2 dependencies are currently used in `:polaris-service`, but `:polaris-service` also contains code needed for Quarkus, which causes conflicts.

Specific implementations, DW + Quarkus, should live in separate modules, ideally underneath separate folders in the code base, for example `quarkus/polaris-quarkus-service` or `dropwizard/polaris-dropwizard-service`, to group additional  framework specific modules together.

### PR DESCRIPTION
**Pull request by adutra**
Fixes #521.

This PR simply splits `polaris-service` in two modules: `polaris-service-common` and `polaris-service-dropwizard`, the former bearing no dependency to Dropwizard.

99% of the changes are file moves, the rest being minor cleanup.

The tests were all moved to `polaris-service-dropwizard` for now. We need a way to factorize them, but let's tackle that later as it's a complex task.

A Quarkus runtime will be provided shortly. It will require some more adjustments to `polaris-service-common`, but again, let's proceed step by step.

### CODE REVIEW

**Review by snazy — COMMENTED** [review]
Big, but "boring" PR ;)

Generally LGTM!

Just would like to have the dropwizard specific stuff in its own top-level directory outside of service:

```
$POLARIS_ROOT/
   dropwizard/service/build.gradle.kts
   service/common/build.gradle.kts
``` 
This allows us to have modules per concern for the frameworks and for the services. I think that keeps things cleaner in the long run.

File: dropwizard/service/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    mergeServiceFiles()
+    append("META-INF/hk2-locator/default")
+    isZip64 = true
+    finalizedBy("startScripts")
+  }
+
+val startScripts =
+  tasks.named<CreateStartScripts>("startScripts") {
+    classpath = files(provider { shadowJar.get().archiveFileName })
+    applicationName = "polaris-service"
```
Comment (adutra) [inline dropwizard/service/build.gradle.kts:175]: This is required to keep the script name to `polaris-service` and the environment variable to `$POLARIS_SERVICE_OPTS`

File: .dockerignore [hunk 2]
```diff
@@ -17,8 +17,6 @@
 # under the License.
 #
 
-polaris-service/logs
-polaris-service/build
-polaris-core/build
+logs
```
Comment (snazy) [inline .dockerignore:20]: is this correct? shouldn't it be dropwizard/service/logs ?
similarly for the `*/build` folders

File: dropwizard/service/build.gradle.kts (hunk 1 above)
Comment (snazy) [reply dropwizard/service/build.gradle.kts:175]: Do we want to keep it as `polaris-service` or change it to `polaris-dropwizard-service` as well?
IMO the latter is likely better to not confuse people w/ Quarkus later.

File: service/common/src/main/java/org/apache/polaris/service/auth/Authenticator.java [hunk 3]
```diff
@@ -16,10 +16,12 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.polaris.service.config;
+package org.apache.polaris.service.auth;
```
Comment (snazy) [inline service/common/src/main/java/org/apache/polaris/service/auth/Authenticator.java:19]: :+1:

File: site/content/in-dev/unreleased/metastores.md [hunk 4]
```diff
@@ -76,7 +76,7 @@ A single `persistence.xml` can describe multiple [persistence units](https://ecl
 To build Polaris with the necessary H2 dependency and start the Polaris service, run the following:
 ```bash
 polaris> ./gradlew --no-daemon --info -PeclipseLink=true -PeclipseLinkDeps=com.h2database:h2:2.3.232 clean shadowJar
-polaris> java -jar  polaris-service/build/libs/polaris-service-*.jar server ./polaris-server.yml
+polaris> java -jar dropwizard/service/build/libs/polaris-service-*.jar server ./polaris-server.yml
```
Comment (snazy) [inline site/content/in-dev/unreleased/metastores.md:79]:
```suggestion
polaris> java -jar dropwizard/service/build/libs/polaris-dropwizard-service-*.jar server ./polaris-server.yml
```
?

File: site/content/in-dev/unreleased/metastores.md [hunk 5]
```diff
@@ -108,5 +108,5 @@ The following shows a sample configuration for integrating Polaris with Postgres
 To build Polaris with the necessary Postgres dependency and start the Polaris service, run the following:
 ```bash
 polaris> ./gradlew --no-daemon --info -PeclipseLink=true -PeclipseLinkDeps=org.postgresql:postgresql:42.7.4 clean shadowJar
-polaris> java -jar  polaris-service/build/libs/polaris-service-*.jar server ./polaris-server.yml
+polaris> java -jar dropwizard/service/build/libs/polaris-service-*.jar server ./polaris-server.yml
```
Comment (snazy) [inline site/content/in-dev/unreleased/metastores.md:111]:
```suggestion
polaris> java -jar dropwizard/service/build/libs/polaris-dropwizard-service-*.jar server ./polaris-server.yml
```
?

Comment (snazy) [inline site/content/in-dev/unreleased/metastores.md:111]: Related: should the config file be renamed to `polaris-dropwizard-service.yml`, too?

File: .dockerignore (hunk 2 above)
Comment (adutra) [reply .dockerignore:20]: If I add e.g. `dropwizard/service/build` IntelliJ warns me: 

> 'dropwizard/service/build' is covered by 'build'

So, I went for simplicity :-)

**Review by eric-maynard — APPROVED** [review]

**Review by snazy — APPROVED** [review]

File: .dockerignore (hunk 2 above)
Comment (dimas-b) [reply .dockerignore:20]: plain `logs` applies at any nesting level, AFAIK

**Review by dimas-b — APPROVED** [review]

**Comment by collado-mike** [comment]
I was hoping to avoid doing this altogether as I thought we had agreed on moving forward with the Quarkus approach in the very near future. At most, I hope this PR is short-lived so that we can continue to have the service implementation and all of its tests in the same gradle package.

**Comment by dimas-b** [comment]
@collado-mike : I think it is valuable to isolate services code from runtime-specific code in any case (Quarkus or DW or both). Integration tests have to depend on the runtime, but could also be reused by downstream project where people want to package additional code into the server (by reusing the services module, yet providing custom "application" code), but make sure the functionality is not broken.

**Pull request merged**